server/channels/app/platform/license.go

```diff
@@ -344,3 +353,33 @@ func (ps *PlatformService) RequestTrialLicense(trialRequest *model.TrialLicenseR
 func (ps *PlatformService) getRequestTrialURL() string {
 	return fmt.Sprintf("%s/api/v1/trials", *ps.Config().CloudSettings.CWSURL)
 }
+
+func (ps *PlatformService) logLicense(message string, license *model.License) {
+	if ps.logger == nil {
```

Comment:
Should we also guard here for a nil  `license`?  I see it's done before in your current calls but maybe we should pull it in for future callers

Reply:
I'm torn. I only added any of the `nil` checks because our tests don't consistently setup the `PlatformService`. If `model.License` was smaller, I'd probably just pass it by value and skip the pointer (since we do a `nil` check already in the caller), but historically throughout the code we haven't trusted the Go compiler on this front and default to pointers almost always. If I do a `nil` check here, I'd either have to just do nothing (which might be equally surprising to the caller) or change the signature to return an `error`.

@agnivade, I'm rusty on best practices here. If a function accepts a non-optional parameter that happens to be passed as a pointer, is it on the caller to do the right thing or the callee?

Reply:
It should be the caller. But we do have a tendency in our codebase to slap on additional checks on the callee to take care of "what if" scenarios. My suggestion there has always been to - "add a unit test".

Anyways, I would say the code here could be improved. The entire confusion stems from the fact that we are calling "SetLicense" once with a license, and another time without a license, which actually removes the license.

It should simply be a separate method! "RemoveLicense" should wipe off the license and "SetLicense" should just set it. That way you remove any confusion of having to add nil checks. Of course, someone can again call "SetLicense" with a nil license, but we have code review for that.

Just my thoughts :)

Reply:
Thanks, @agnivade! I was tempted to start improving this service but very quickly found myself second guessing as to unexpected side effects. (I ripped out a number of changes from this PR before submitting).

Given I think Maria's feedback is about the new `logLicense` specifically, I'll propose leaving it to the caller (all of whom do the right thing, logging either the new license of the old one just removed).

Thank you!